confluent_kafka/src/Producer.c

```diff
@@ -361,10 +364,12 @@ static PyObject *Producer_produce (Handle *self, PyObject *args,
 	if (!partitioner_cb || partitioner_cb == Py_None)
 		partitioner_cb = self->u.Producer.partitioner_cb;
 
+#ifdef RD_KAFKA_V_HEADERS
```

Comment:
You actually want to bail out with an exception if headers is non-NULL and #ifndef RD_KAFKA_V_HEADERS.
See same thing for timestamp above.